spec/bundler/dsl_spec.rb

```diff
@@ -144,6 +144,11 @@
       expect { subject.gem(:foo) }.
         to raise_error(Bundler::GemfileError, /You need to specify gem names as Strings. Use 'gem "foo"' instead/)
     end
+
+    it "rejects branch option on non-git gems" do
```

Comment:
please also test that specifying a branch when using a user-defined git source still passes? (or make sure such a test exists?)